Source/WebKit/UIProcess/API/haiku/WebView.cpp

```diff
@@ -70,31 +70,15 @@ BWebView::BWebView(BRect frame, BWindow* myWindow)
 
 void BWebView::navigationCallbacks()
 {
-    auto page = WKViewGetPage(fViewPort.get());
-    WKPageNavigationClientV0 navigationClient = {};
+    fWebViewBase->page()->setNavigationClient(makeUniqueRef<NavigationClient>(this));
 
-    navigationClient.base.version = 0;
-    navigationClient.base.clientInfo = this;
-
-    navigationClient.didCommitNavigation = didCommitNavigation;
-    navigationClient.didFinishDocumentLoad = didFinishDocumentLoad;
-    navigationClient.didFailNavigation = didFailNavigation;
-    navigationClient.didFinishNavigation = didFinishNavigation;
-    navigationClient.didReceiveServerRedirectForProvisionalNavigation = didReceiveServerRedirectForProvisionalNavigation;
-    WKPageSetPageNavigationClient(page, &navigationClient.base);
-
-    fObserver = new PageLoadStateObserver(fAppLooper);
-    getRenderView()->page()->pageLoadState().addObserver(*fObserver);
+    fObserver = makeUnique<PageLoadStateObserver>(fAppLooper);
+    fWebViewBase->page()->pageLoadState().addObserver(*fObserver);
 }
 
 void BWebView::loadHTML()
 {
-    auto page = WKViewGetPage(fViewPort.get());
-    WKRetainPtr<WKURLRef> uri;
-    uri = adoptWK(WKURLCreateWithUTF8CString("about:blank"));
-    WKRetainPtr<WKStringRef> str;
-    str = adoptWK(WKStringCreateWithUTF8CString("<body>Hello world</body>"));
-    WKPageLoadHTMLString(page, str.get(), uri.get());
+    fWebViewBase->page()->loadData("<body>Hello world</body>"_s.span8(), "text/html"_s, "latin1"_s, "about:blank"_s, nullptr);
```

Comment:
why did this change from UTF-8 to Latin-1? (not that it makes much of a difference)